### PR TITLE
advanced edit

### PR DESCRIPTION
**Pull request by theseyi**

### CODE REVIEW

File: wherehows-web/app/components/dataset-compliance.ts [hunk 1]
```diff
@@ -902,6 +970,56 @@ export default class DatasetCompliance extends Component {
...
+        };
+        const isValid = validateMetadataObject(metadataObject, complianceEntitiesTaxonomy);
+
+        set(this, 'isManualApplyDisabled', !isValid);
+
+        if (isValid) {
+          set(this, 'manuallyEnteredComplianceEntities', metadataObject);
+        }
+      } catch {
+        set(this, 'isManualApplyDisabled', true);
```
Comment (cptran777) [inline wherehows-web/app/components/dataset-compliance.ts]: Would it be helpful to catch what kind of error we are getting here in order to help the user understand why this is disabled? Like JSON.parse returning an error would be a different cue than `validateMetadataObject()` returning an error I'd assume

File: wherehows-web/app/services/notifications.ts [hunk 2]
```diff
@@ -152,7 +153,7 @@ const notificationHandlers: INotificationHandler = {
     };
     // Set default values for button text if none are provided by consumer
     props = { dismissButtonText: 'No', confirmButtonText: 'Yes', ...props };
-    const { dismissButtonText, confirmButtonText } = props;
+    const { dismissButtonText, confirmButtonText, onDialogToggle } = props;
```
Comment (cptran777) [inline wherehows-web/app/services/notifications.ts]: Maybe it's the react in me speaking, but I thought `props` was like passed in component properties. hahaha

File: wherehows-web/app/components/dataset-compliance.ts (hunk 1 above)
Comment (theseyi) [reply wherehows-web/app/components/dataset-compliance.ts]: 👍 yea, that's an enhancement to be worked on, for the first case, right now if the JSON is invalid, the editor shows an error indicator on the line where the issue occurred.
For the other case, `validateMetadataObject` is a boolean operation atm, the enhancement would be to fix this by capturing an propagating the issue that caused the validation to fail and showing that to the user

File: wherehows-web/app/services/notifications.ts (hunk 2 above)
Comment (theseyi) [reply wherehows-web/app/services/notifications.ts]: haha, react? what's that?, jk not in this case :)

**Pull request merged**